CMakeLists.txt

```diff
@@ -102,17 +100,21 @@ ffig_add_library(NAME Tree INPUTS tests/input/Tree.h NOEXCEPT PYTHON CPP D SWIFT
 ffig_add_library(NAME Asset INPUTS tests/input/Asset.h PYTHON LUA)
 ffig_add_library(NAME Animal INPUTS tests/input/Animal.h PYTHON)
 
-# build dotnet test dll
-# FIXME: Work out how to get dependencies calculated more robustly
 if(dotnet_FOUND)
-  file(GLOB TEST_INPUT ${CMAKE_SOURCE_DIR}/tests/input/*.h)
-  add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/generated/ffig.net.dll
-    COMMAND dotnet build ffig.net.csproj -o .
-    DEPENDS ${CMAKE_BINARY_DIR}/generated/Shape.cs;${CMAKE_BINARY_DIR}/generated/Tree.cs
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/generated
-    COMMENT "Building DOTNET assembly ffig.net.dll")
-  add_custom_target(ffig.net ALL
-    DEPENDS ${CMAKE_BINARY_DIR}/generated/ffig.net.dll)
+  include(dotnet)
+  add_dotnet_library(NAME ffig.net 
+    DIRECTORY ${CMAKE_BINARY_DIR}/generated
+    SOURCES 
+      ${CMAKE_SOURCE_DIR}/tests/dotnet/TestShape.cs 
```

Comment:
In general, CMAKE_CURRENT_LIST_DIR and CMAKE_CURRENT_BINARY_DIR are prefered to allow relocating the various parts to different files and avoid conflicts with output files.

Reply:
Fixed.

Reply:
I'll defer making this change to CMakeLists.txt to another PR to avoid conflating cleanup with dotnet additions.